pkg/search/postgres/store.go

```diff
@@ -263,3 +263,33 @@ func (s *GenericStore[T, PT]) GetByQuery(ctx context.Context, query *v1.Query) (
 	}
 	return rows, nil
 }
+
+// GetIDs returns all the IDs for the store.
+func (s *GenericStore[T, PT]) GetIDs(ctx context.Context) ([]string, error) {
+	defer s.setPostgresOperationDurationTime(time.Now(), ops.GetAll)
+	var sacQueryFilter *v1.Query
+	if s.hasPermissionsChecker() {
+		if ok, err := s.permissionChecker.GetAllowed(ctx); err != nil {
+			return nil, err
+		} else if !ok {
+			return nil, sac.ErrResourceAccessDenied
```

Comment:
This changes the behaviour compared to the generated code.
I'd stick to the previous behaviour that returns `nil, nil`, otherwise, this may leak information about the user allowed scope.